scripts/extract_schemas.py

```
@@ -0,0 +1,142 @@
+            return yaml.safe_load(f)
+    except yaml.YAMLError as e:
+        print(f"Error: Failed to parse YAML in {SPEC_PATH}: {e}", file=sys.stderr)
+        sys.exit(1)
+
+
+def save_spec(spec):
+    with open(SPEC_PATH, "w") as f:
+        yaml.dump(spec, f, default_flow_style=False, sort_keys=False, allow_unicode=True)
+
+
+def _safe_extract(spec, path_keys, schema_name, item_key="items"):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_safe_extract`

Add return type annotation: `None`

(ANN202)